### PR TITLE
Add a recommendation to set process resource attributes

### PR DESCRIPTION
**Pull request by pellared**
## Why

Maintenance 😉  These resource attributes are already set by our distributions.

## What 

Add a recommendation to set process resource attributes.

### CODE REVIEW

File: specification/semantic_conventions.md [hunk 1]
```diff
@@ -26,6 +26,8 @@ All Splunk distributions of OpenTelemetry,
 - SHOULD set the following resource attributes when applicable:
   - `telemetry.auto.version`
 
+- SHOULD set [process an process runtime resource attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.18.0/specification/resource/semantic_conventions/process.md)
```
Comment (pellared) [inline specification/semantic_conventions.md:29]: note: The OTel spec is "experimental". However, our distros are already setting these attributes. That is why I put a `SHOULD`. Still I am unsure if this should land in "stable" section or we should it be under "experimental".

Comment (pellared) [reply specification/semantic_conventions.md:29]: Now I see that even https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#telemetry-sdk is marked as `Experimental`...

Comment (theletterf) [reply specification/semantic_conventions.md:29]: `process an process` (is that right?)

Comment (pellared) [reply specification/semantic_conventions.md:29]: typo fixed 😆 it should be `and`.

**Review by breedx-splk — APPROVED** [review]

**Pull request merged**